[.NET] Bundled Copilot Cli download times out

## Steps to repro

Follow the getting started guide.

0. macOS 26 (Arm64)
1. `dotnet new console -n $whatever -o $wherever`
2. Add to .csproj:
   ```xml
   <ItemGroup Label="Dependencies">
     <PackageReference Include="GitHub.Copilot.SDK" Version="0.1.24-preview.0" />
   </ItemGroup>
   ```
3. Copy-paste in the example code from [getting started guide (permalink)](https://github.com/github/copilot-sdk/blob/4dc562951a51097618a69a1837f7af06c73d1113/docs/getting-started.md)
4. Run `dotnet run -v diag`

## Error message:

```
CopilotPlayground % dotnet run -v diag
Restore complete (0.2s)
    Determining projects to restore...
    All projects are up-to-date for restore.
  CopilotPlayground net10.0 failed with 1 error(s) (100.1s)
    Downloading Copilot CLI 0.0.405 for darwin-arm64...
    /Users/loganbussell/.nuget/packages/github.copilot.sdk/0.1.24-preview.0/build/GitHub.Copilot.SDK.targets(47,5): error MSB3923: Failed to download file "https://registry.npmjs.org/@github/copilot-darwin-arm64/-/copilot-darwin-arm64-0.0.405.tgz".  The request was canceled due to the configured HttpClient.Timeout of 100 seconds elapsing. ---> A task was canceled. ---> A task was canceled.

Build failed with 1 error(s) in 100.5s

The build failed. Fix the build errors and run again.
```

When I download the tarball from the link using my browser, it works no problem (instantly) https://registry.npmjs.org/@github/copilot-darwin-arm64/-/copilot-darwin-arm64-0.0.405.tgz.

--- a/dotnet/src/build/GitHub.Copilot.SDK.targets
+++ b/dotnet/src/build/GitHub.Copilot.SDK.targets
@@ -48,6 +48,13 @@
     <CopilotNpmRegistryUrl Condition="'$(CopilotNpmRegistryUrl)' == ''">https://registry.npmjs.org</CopilotNpmRegistryUrl>
   </PropertyGroup>
 
+  <!-- Timeout in seconds for downloading the Copilot CLI tarball.
+       Override CopilotCliDownloadTimeout in your .csproj or Directory.Build.props if needed.
+       Default is 600 seconds (10 minutes) to handle slow or unreliable network conditions. -->
+  <PropertyGroup>
+    <CopilotCliDownloadTimeout Condition="'$(CopilotCliDownloadTimeout)' == ''">600</CopilotCliDownloadTimeout>
+  </PropertyGroup>
+
   <!-- Download and extract CLI binary -->
   <Target Name="_DownloadCopilotCli" BeforeTargets="BeforeBuild" Condition="'$(_CopilotPlatform)' != ''">
     <Error Condition="'$(CopilotCliVersion)' == ''" Text="CopilotCliVersion is not set. The GitHub.Copilot.SDK.props file may be missing from the NuGet package." />
@@ -68,6 +75,7 @@
     <MakeDir Directories="$(_CopilotCacheDir)" Condition="!Exists('$(_CopilotCliBinaryPath)')" />
     <Message Importance="high" Text="Downloading Copilot CLI $(CopilotCliVersion) for $(_CopilotPlatform)..." Condition="!Exists('$(_CopilotCliBinaryPath)')" />
     <DownloadFile SourceUrl="$(_CopilotDownloadUrl)" DestinationFolder="$(_CopilotCacheDir)" DestinationFileName="copilot.tgz"
+                  Timeout="$([System.Int32]::Parse('$(CopilotCliDownloadTimeout)'))"
                   Condition="!Exists('$(_CopilotCliBinaryPath)')" />
 
     <!-- Extract using tar (use Windows system tar explicitly to avoid Git bash tar issues) -->

